core-plugins/src/main/java/co/cask/hydrator/plugin/common/KinesisUtil.java

```diff
@@ -0,0 +1,159 @@
+          return;
+        default:
+          throw new IllegalStateException("Illegal stream state: " + (state.equals("") ? "State not found" : state));
+      }
+    } else {
+      createStream(streamName, shardCount, kinesisClient);
+    }
+    long waitTimeCreate = System.currentTimeMillis() + CREATE_TIMEOUT;
+    while (System.currentTimeMillis() < waitTimeCreate) {
+      try {
+        Thread.sleep(SLEEP_INTERVAL);
+      } catch (InterruptedException e) {
+        Thread.currentThread().interrupt();
+      }
+      try {
+        String streamStatus = getStreamState(kinesisClient, streamName);
+        if (streamStatus.equals(STATUS_ACTIVE)) {
+          LOG.info("Stream {} is ACTIVE", streamName);
+          return;
+        }
+      } catch (ResourceNotFoundException e) {
+        throw new IllegalStateException(String.format("Stream %s did not become active in %d",
+                                                      streamName, CREATE_TIMEOUT), e);
+      }
+    }
+    if (!getStreamState(kinesisClient, streamName).equals(STATUS_ACTIVE)) {
+      throw new IllegalStateException(String.format("Stream %s did not become active in %d",
+                                                    streamName, CREATE_TIMEOUT));
+    }
+  }
+
+  private static void createStream(String streamName, int shardCount, AmazonKinesisClient kinesisClient) {
+    CreateStreamRequest createStreamRequest = new CreateStreamRequest();
+    createStreamRequest.setStreamName(streamName);
+    createStreamRequest.setShardCount(shardCount);
+    kinesisClient.createStream(createStreamRequest);
+    LOG.info("Stream " + streamName + " created");
+  }
+
+  /**
+   * Return the state of a Amazon Kinesis stream.
+   *
+   * @param kinesisClient The {@link AmazonKinesisClient} with Amazon Kinesis read privileges
+   * @param streamName The Amazon Kinesis stream to get the state of
+   * @return String representation of the Stream state
+   */
+  private static String getStreamState(AmazonKinesisClient kinesisClient, String streamName) {
+    DescribeStreamRequest describeStreamRequest = new DescribeStreamRequest();
+    describeStreamRequest.setStreamName(streamName);
+    try {
+      return kinesisClient.describeStream(describeStreamRequest).getStreamDescription().getStreamStatus();
+    } catch (AmazonServiceException e) {
+      LOG.debug("State of the stream {} could not be found", streamName);
+      return "";
+    }
+  }
+
+  /**
+   * Helper method to determine if an Amazon Kinesis stream exists.
+   *
+   * @param kinesisClient The {@link AmazonKinesisClient} with Amazon Kinesis read privileges
+   * @param streamName The Amazon Kinesis stream to check for
+   * @return true if the Amazon Kinesis stream exists, otherwise return false
+   */
+  private static boolean streamExists(AmazonKinesisClient kinesisClient, String streamName) {
+    DescribeStreamRequest describeStreamRequest = new DescribeStreamRequest();
```

Comment:
instead of doing this all over again, can you refactor to just call the above method from here and do appropriate exception handling?